packages/flutter_tools/lib/src/base/logger.dart

```diff
@@ -417,12 +430,18 @@ class AnsiStatus extends AnsiSpinner {
   /// milliseconds.  Only backs up 4 characters because [super.cancel] backs
   /// up one.
   ///
+  /// If [multiLineOutput] is true, then it prints the message again on a new line
+  /// before writing the elapsed time.
+  ///
   /// Example: '\b\b\b\b 0.5s', '\b\b\b\b150ms', '\b\b\b\b1600ms'
   void writeSummaryInformation() {
+    final String restateMessage = multiLineOutput
+        ? '    \n${'$message Done'.padRight(padding - 4)}$_margin'
+        : '';
     if (expectSlowOperation) {
-      stdout.write('\b\b\b\b${getElapsedAsSeconds(stopwatch.elapsed).padLeft(5)}');
+      stdout.write('\b\b\b\b$restateMessage${getElapsedAsSeconds(stopwatch.elapsed).padLeft(5)}');
     } else {
-      stdout.write('\b\b\b\b${getElapsedAsMilliseconds(stopwatch.elapsed).padLeft(5)}');
+      stdout.write('\b\b\b\b$restateMessage${getElapsedAsMilliseconds(stopwatch.elapsed).padLeft(5)}');
```

Comment:
if it's multiline output, surely we shouldn't be using AnsiStatus at all? since it'll be backspacing over random characters?

Reply:
Well, yeah.  But we won't get a progress spinner then either, since without backspacing, we won't be able to animate.

Maybe the right answer is to not print the `\b`'s before the `$restateMessage`.  That way we won't backspace over things that have been printed in the interim, but still get to animate.  There is some chance of the spinner itself messing with the output, but that's true already, and doesn't happen that often in practice.

One thing that might help is to switch the spinner to use something like `['⠋', '⠙', '⠹', '⠸', '⠼', '⠴', '⠦', '⠧', '⠇', '⠏']` as the spinner instead so that it's less likely to be confused with regular output.  We could also color it.